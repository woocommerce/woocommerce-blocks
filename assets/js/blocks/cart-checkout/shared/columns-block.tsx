/**
 * External dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';

export const Columns = ( {
	children,
	...props
}: {
	children?: React.ReactNode;
} ): JSX.Element => {
	const blockProps = useBlockProps( props );

	return <div { ...blockProps }>{ children }</div>;
};
