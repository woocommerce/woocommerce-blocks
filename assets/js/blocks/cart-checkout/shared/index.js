export * from './hacks';
export * from './use-forced-layout';
export * from './editor-utils';
export * from './use-view-switcher';
