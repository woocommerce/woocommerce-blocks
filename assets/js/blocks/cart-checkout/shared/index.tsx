export * from './hacks';
export * from './editor-utils';
export * from './use-forced-layout';
