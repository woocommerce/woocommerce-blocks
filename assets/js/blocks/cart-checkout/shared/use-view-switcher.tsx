/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { useState } from '@wordpress/element';
import { useDispatch, select } from '@wordpress/data';
import { ToolbarGroup, ToolbarDropdownMenu } from '@wordpress/components';
import { Icon, eye } from '@woocommerce/icons';
import { store as blockEditorStore } from '@wordpress/block-editor';

interface View {
	view: string;
	label: string;
	icon: string | JSX.Element;
}

export const useViewSwitcher = (
	clientId: string,
	views: View[]
): {
	currentView: string;
	component: JSX.Element;
} => {
	const initialView = views[ 0 ];
	const [ currentView, setCurrentView ] = useState( initialView );
	const { selectBlock } = useDispatch( 'core/block-editor' );
	const { getBlock } = select( blockEditorStore );

	const ViewSwitcherComponent = (
		<ToolbarGroup>
			<ToolbarDropdownMenu
				label={ __( 'Switch view', 'woo-gutenberg-products-block' ) }
				text={ currentView.label }
				icon={
					<Icon srcElement={ eye } style={ { marginRight: '8px' } } />
				}
				controls={ views.map( ( view ) => ( {
					...view,
					title: <span>{ view.label }</span>,
					isActive: view.view === currentView.view,
					onClick: () => {
						setCurrentView( view );
						selectBlock(
							getBlock( clientId ).innerBlocks.find(
								( block: { name: string } ) =>
									block.name === view.view
							)?.clientId || clientId
						);
					},
				} ) ) }
			/>
		</ToolbarGroup>
	);

	return {
		currentView: currentView.view,
		component: ViewSwitcherComponent,
	};
};
