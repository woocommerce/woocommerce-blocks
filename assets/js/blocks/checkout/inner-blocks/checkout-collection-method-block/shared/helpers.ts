/**
 * External dependencies
 */
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

/**
 * Returns the cheapest rate that isn't a local pickup.
 *
 * @param {Array|undefined} shippingRates Array of shipping Rate.
 *
 * @return {Object|undefined} cheapest rate.
 */
export function getShippingStartingPrice(
	shippingRates: CartShippingPackageShippingRate[]
): CartShippingPackageShippingRate | undefined {
	if ( shippingRates ) {
		return shippingRates.reduce(
			(
				lowestRate: CartShippingPackageShippingRate | undefined,
				currentRate: CartShippingPackageShippingRate
			) => {
				if ( currentRate.method_id === 'local_pickup' ) {
					return lowestRate;
				}
				if (
					lowestRate === undefined ||
					currentRate.price < lowestRate.price
				) {
					return currentRate;
				}
				return lowestRate;
			},
			undefined
		);
	}
}

/**
 * Returns the cheapest rate that is a local pickup.
 *
 * @param {Array|undefined} shippingRates Array of shipping Rate.
 *
 * @return {Object|undefined} cheapest rate.
 */
export function getLocalPickupStartingPrice(
	shippingRates: CartShippingPackageShippingRate[]
): CartShippingPackageShippingRate | undefined {
	if ( shippingRates ) {
		return shippingRates.reduce(
			(
				lowestRate: CartShippingPackageShippingRate | undefined,
				currentRate: CartShippingPackageShippingRate
			) => {
				if ( currentRate.method_id !== 'local_pickup' ) {
					return lowestRate;
				}
				if (
					lowestRate === undefined ||
					currentRate.price < lowestRate.price
				) {
					return currentRate;
				}
				return lowestRate;
			},
			undefined
		);
	}
}
