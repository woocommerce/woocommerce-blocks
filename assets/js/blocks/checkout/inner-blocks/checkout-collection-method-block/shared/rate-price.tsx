/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { getSetting } from '@woocommerce/settings';
import { createInterpolateElement } from '@wordpress/element';
import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

export const RatePrice = ( {
	minRate,
	maxRate,
}: {
	minRate: CartShippingPackageShippingRate | undefined;
	maxRate: CartShippingPackageShippingRate | undefined;
} ) => {
	if ( ! minRate && ! maxRate ) {
		return null;
	}
	const minRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( minRate.price, 10 ) + parseInt( minRate.taxes, 10 )
		: parseInt( minRate.price, 10 );
	const maxRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( maxRate.price, 10 ) + parseInt( maxRate.taxes, 10 )
		: parseInt( maxRate.price, 10 );

	return (
		<span className="wc-block-checkout__collection-item-price">
			{ minRatePrice === 0 && maxRatePrice === 0 ? (
				<em>{ __( 'free', 'woo-gutenberg-products-block' ) }</em>
			) : (
				createInterpolateElement(
					__( 'from <price />', 'woo-gutenberg-products-block' ),
					{
						price:
							minRatePrice === 0 ? (
								<em>
									{ __(
										'free',
										'woo-gutenberg-products-block'
									) }
								</em>
							) : (
								<FormattedMonetaryAmount
									currency={ getCurrencyFromPriceResponse(
										minRate
									) }
									value={ minRatePrice }
								/>
							),
					}
				)
			) }
		</span>
	);
};
