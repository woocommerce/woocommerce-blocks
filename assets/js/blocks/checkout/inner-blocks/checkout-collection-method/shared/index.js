export { RatePrice } from './rate-price';
export * from './helpers';
