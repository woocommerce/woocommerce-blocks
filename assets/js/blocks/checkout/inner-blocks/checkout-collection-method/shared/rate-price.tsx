/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import { getSetting } from '@woocommerce/settings';
import { createInterpolateElement } from '@wordpress/element';
import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';

export const RatePrice = ( {
	rate,
}: {
	rate: CartShippingPackageShippingRate;
} ) => {
	const ratePrice = getSetting( 'displayCartPricesIncludingTax', false )
		? parseInt( rate.price, 10 ) + parseInt( rate.taxes, 10 )
		: parseInt( rate.price, 10 );
	return (
		<span className="wc-block-checkout__collection-item-price">
			{ ratePrice === 0
				? __( 'free', 'woo-gutenberg-products-block' )
				: createInterpolateElement(
						__( 'from <price />', 'woo-gutenberg-products-block' ),
						{
							price: (
								<FormattedMonetaryAmount
									currency={ getCurrencyFromPriceResponse(
										rate
									) }
									value={ ratePrice }
								/>
							),
						}
				  ) }
		</span>
	);
};
