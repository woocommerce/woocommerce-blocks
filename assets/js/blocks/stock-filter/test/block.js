/**
 * Internal dependencies
 */

/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { default as fetchMock } from 'jest-fetch-mock';

/**
 * Internal dependencies
 */
import Block from '../block';
import { allSettings } from '../../../settings/shared/settings-init';

const mockResults = {
	stock_status_counts: [
		{ status: 'instock', count: '18' },
		{ status: 'outofstock', count: '1' },
		{ status: 'onbackorder', count: '5' },
	],
};

jest.mock( '@woocommerce/base-context/hooks', () => {
	return {
		...jest.requireActual( '@woocommerce/base-context/hooks' ),
		useCollectionData: () => ( { isLoading: false, results: mockResults } ),
	};
} );

const StockFilterBlock = ( props ) => <Block { ...props } />;
describe( 'Testing stock filter', () => {
	beforeEach( () => {
		allSettings.stockStatusOptions = {
			instock: 'In stock',
			outofstock: 'Out of stock',
			onbackorder: 'On backorder',
		};
	} );

	afterEach( () => {
		fetchMock.resetMocks();
	} );

	it( 'renders the stock filter block', async () => {
		const { container } = render(
			<StockFilterBlock attributes={ { isPreview: false } } />
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'renders the stock filter block with the filter button', async () => {
		const { container } = render(
			<StockFilterBlock
				attributes={ { isPreview: false, showFilterButton: true } }
			/>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'renders the stock filter block with the product counts', async () => {
		const { container } = render(
			<StockFilterBlock
				attributes={ {
					isPreview: false,
					showCounts: true,
				} }
			/>
		);
		expect( container ).toMatchSnapshot();
	} );
} );
