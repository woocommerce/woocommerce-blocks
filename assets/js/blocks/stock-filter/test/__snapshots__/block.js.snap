// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing stock filter renders the stock filter block 1`] = `
<div>
  <div
    class="wc-block-stock-filter"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="instock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="instock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="onbackorder"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="onbackorder"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="outofstock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="outofstock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  />
</div>
`;

exports[`Testing stock filter renders the stock filter block with the filter button 1`] = `
<div>
  <div
    class="wc-block-stock-filter"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="instock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="instock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="onbackorder"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="onbackorder"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="outofstock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="outofstock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  >
    <button
      class="wp-block-button__link wc-block-filter-submit-button wc-block-components-filter-submit-button wc-block-stock-filter__button"
      type="submit"
    >
      <span
        aria-hidden="true"
      >
        Apply
      </span>
      <span
        class="screen-reader-text"
      >
        Apply filter
      </span>
    </button>
  </div>
</div>
`;

exports[`Testing stock filter renders the stock filter block with the product counts 1`] = `
<div>
  <div
    class="wc-block-stock-filter"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="instock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="instock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  18
                </span>
                <span
                  class="screen-reader-text"
                >
                  18 products
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="onbackorder"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="onbackorder"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  5
                </span>
                <span
                  class="screen-reader-text"
                >
                  5 products
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="outofstock"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="outofstock"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  1
                </span>
                <span
                  class="screen-reader-text"
                >
                  1 product
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  />
</div>
`;
